enterprise/backend/src/metabase_enterprise/serialization/v2/storage.clj

```diff
@@ -1,15 +1,45 @@
 (ns metabase-enterprise.serialization.v2.storage
-  "A multimethod entry point for storage sinks. Storage is the second phase of serialization.
-  See [[metabase.models.serialization.base]] for detailed documentation of the serialization process.
-  Implementations of storage should live in [[metabase-enterprise.serialization.v2.storage.yaml]] and similar.")
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str]
+            [metabase-enterprise.serialization.dump :refer [spit-yaml]]
+            [metabase.models.serialization :as serdes]
+            [metabase.util.i18n :refer [trs]]
+            [metabase.util.log :as log]))
 
-(defmulti store-all!
-  "`(store-all! stream opts)`
-  `stream` is a reducible stream of portable maps with `:serdes/meta` keys.
-  `opts` is a map of options, such as the path to the root directory.
+(defn- escape-segment
```

Comment:
Since the filesystem path is now really just a convenience for the user, we could probably just remove slashes and backslashes, or do something else. This can be tricky due to unicode in labels. Thoughts @bshepherdson ?